### PR TITLE
Enable "gosimple" linter plugin

### PR DESCRIPTION
**Pull request by mboersma**
**What this PR does / why we need it**:

Simplifies some code as suggested by the `gosimple` plugin to `gometalinter` and enables that plugin for the `make test-style` target.

~(We can't run this plugin in general for linting, however, because it suggests some incorrect changes to `pkg/api/types_test.go`.)~

**Special notes for your reviewer**:

Obviously not a critical change, but I've had this branch sitting around for a month so I freshened it up and made a PR.

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/acsengine/defaults_test.go [hunk 1]
```diff
@@ -618,12 +618,12 @@ func TestSetVMSSDefaults(t *testing.T) {
 
 	properties.AgentPoolProfiles[0].Count = 110
 	setPropertiesDefaults(&mockCS, false, false)
-	if *properties.AgentPoolProfiles[0].SinglePlacementGroup != false {
```
Comment (jackfrancis) [inline pkg/acsengine/defaults_test.go]: lol

File: pkg/acsengine/defaults_test.go [hunk 2]
```diff
@@ -618,12 +618,12 @@ func TestSetVMSSDefaults(t *testing.T) {
...
+	if *properties.AgentPoolProfiles[0].SinglePlacementGroup {
```
Comment (jackfrancis) [inline pkg/acsengine/defaults_test.go]: Do we need the `*` syntax here at all?

File: pkg/acsengine/transform/apimodel_merger.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ type APIModelValue struct {
 
 // MapValues converts an arraw of rwa ApiModel values (like ["masterProfile.count=4","linuxProfile.adminUsername=admin"]) to a map
 func MapValues(m map[string]APIModelValue, setFlagValues []string) {
-	if setFlagValues == nil || len(setFlagValues) == 0 {
+	if len(setFlagValues) == 0 {
```
Comment (jackfrancis) [inline pkg/acsengine/transform/apimodel_merger.go]: no... really, are you sure you're unset?

File: pkg/api/agentPoolOnlyApi/v20180331/validate.go [hunk 4]
```diff
@@ -40,7 +40,7 @@ func (l *LinuxProfile) Validate() error {
 
 // Validate implements APIObject
 func (a *AADProfile) Validate(rbacEnabled *bool) error {
-	if rbacEnabled == nil || *rbacEnabled == false {
+	if rbacEnabled == nil || !*rbacEnabled {
```
Comment (jackfrancis) [inline pkg/api/agentPoolOnlyApi/v20180331/validate.go]: see `helpers.IsTrueBoolPointer()`, I think that's a more expressive way to put the above. E.g.:

`if !helper.IsTrueBoolPointer(rbacEnabled) {...`

File: pkg/api/converterfromagentpoolonlyapi_test.go [hunk 5]
```diff
@@ -244,7 +244,6 @@ func TestConvertToV20180331AddonProfile(t *testing.T) {
 
 func TestConvertKubernetesConfigToEnableRBACV20180331AgentPoolOnly(t *testing.T) {
 	var kc *KubernetesConfig
-	kc = nil
```
Comment (jackfrancis) [inline pkg/api/converterfromagentpoolonlyapi_test.go]: lol I'm surprised the compiler accepts this

File: pkg/api/convertertoagentpoolonlyapi.go [hunk 6]
```diff
@@ -384,7 +384,7 @@ func convertV20180331AgentPoolOnlyWindowsProfile(obj *v20180331.WindowsProfile)
 }
 
 func convertV20180331AgentPoolOnlyKubernetesConfig(enableRBAC *bool) *KubernetesConfig {
-	if enableRBAC != nil && *enableRBAC == true {
+	if enableRBAC != nil && *enableRBAC {
```
Comment (jackfrancis) [inline pkg/api/convertertoagentpoolonlyapi.go]: see my above suggestion

File: pkg/api/convertertoagentpoolonlyapi_test.go [hunk 7]
```diff
@@ -226,19 +226,18 @@ func TestConvertV20170831AgentPoolOnlyOrchestratorProfile_KubernetesConfig(t *te
 		t.Error("OrchestratorProfile.KubernetesConfig expected not to be nil")
 	}
 
-	if op.KubernetesConfig.EnableRbac == nil || *op.KubernetesConfig.EnableRbac == true {
+	if op.KubernetesConfig.EnableRbac == nil || *op.KubernetesConfig.EnableRbac {
```
Comment (jackfrancis) [inline pkg/api/convertertoagentpoolonlyapi_test.go]: ditto

File: pkg/api/convertertoagentpoolonlyapi_test.go [hunk 8]
```diff
@@ -226,19 +226,18 @@ func TestConvertV20170831AgentPoolOnlyOrchestratorProfile_KubernetesConfig(t *te
...
 		t.Error("OrchestratorProfile.KubernetesConfig.EnableRbac expected to be *false")
 	}
 
-	if op.KubernetesConfig.EnableSecureKubelet == nil || *op.KubernetesConfig.EnableSecureKubelet == true {
+	if op.KubernetesConfig.EnableSecureKubelet == nil || *op.KubernetesConfig.EnableSecureKubelet {
```
Comment (jackfrancis) [inline pkg/api/convertertoagentpoolonlyapi_test.go]: ditto

File: pkg/api/convertertoagentpoolonlyapi_test.go [hunk 9]
```diff
@@ -226,19 +226,18 @@ func TestConvertV20170831AgentPoolOnlyOrchestratorProfile_KubernetesConfig(t *te
...
 		t.Error("OrchestratorProfile.KubernetesConfig.EnableSecureKubelet expected to be *false")
 	}
 
 }
 
 func TestConvertV20180331AgentPoolOnlyKubernetesConfig(t *testing.T) {
-	var kc *KubernetesConfig
-	kc = convertV20180331AgentPoolOnlyKubernetesConfig(helpers.PointerToBool(true))
+	var kc = convertV20180331AgentPoolOnlyKubernetesConfig(helpers.PointerToBool(true))
```
Comment (jackfrancis) [inline pkg/api/convertertoagentpoolonlyapi_test.go]: 💥

File: pkg/api/convertertoagentpoolonlyapi_test.go [hunk 10]
```diff
@@ -272,15 +271,15 @@ func TestConvertV20180331AgentPoolOnlyKubernetesConfig(t *testing.T) {
 		t.Error("EnableRbac expected not to be nil")
 	}
 
-	if *kc.EnableRbac != false {
```
Comment (jackfrancis) [inline pkg/api/convertertoagentpoolonlyapi_test.go]: wow

File: pkg/api/convertertoagentpoolonlyapi_test.go [hunk 11]
```diff
@@ -272,15 +271,15 @@ func TestConvertV20180331AgentPoolOnlyKubernetesConfig(t *testing.T) {
...
+	if *kc.EnableRbac {
 		t.Error("EnableRbac expected to be false")
 	}
 
 	if kc.EnableSecureKubelet == nil {
 		t.Error("EnableSecureKubelet expected not to be nil")
 	}
 
-	if *kc.EnableSecureKubelet != false {
```
Comment (jackfrancis) [inline pkg/api/convertertoagentpoolonlyapi_test.go]: ditto

File: pkg/api/convertertoagentpoolonlyapi_test.go [hunk 12]
```diff
@@ -297,15 +296,15 @@ func TestConvertV20180331AgentPoolOnlyKubernetesConfig(t *testing.T) {
 		t.Error("EnableRbac expected not to be nil")
 	}
 
-	if *kc.EnableRbac != false {
```
Comment (jackfrancis) [inline pkg/api/convertertoagentpoolonlyapi_test.go]: ...

File: pkg/api/convertertoagentpoolonlyapi_test.go [hunk 13]
```diff
@@ -297,15 +296,15 @@ func TestConvertV20180331AgentPoolOnlyKubernetesConfig(t *testing.T) {
...
+	if *kc.EnableRbac {
 		t.Error("EnableRbac expected to be false")
 	}
 
 	if kc.EnableSecureKubelet == nil {
 		t.Error("EnableSecureKubelet expected not to be nil")
 	}
 
-	if *kc.EnableSecureKubelet != false {
```
Comment (jackfrancis) [inline pkg/api/convertertoagentpoolonlyapi_test.go]: make it stop

File: pkg/armhelpers/graph.go [hunk 14]
```diff
@@ -112,7 +112,7 @@ func (az *AzureClient) CreateRoleAssignmentSimple(ctx context.Context, resourceG
 		},
 	}
 
-	re := regexp.MustCompile("(?i)status=(\\d+)")
+	re := regexp.MustCompile(`(?i)status=(\d+)`)
```
[the remark this answers is not in the document]
Comment (mboersma) [reply pkg/armhelpers/graph.go]: The original regex is a normal, interpreted string literal, so it needed to escape the backslash. Using the backtics syntax makes it a [raw string literal](https://golang.org/ref/spec#String_literals), so no escaping is needed. Raw string literals are preferable for regexes for readability and copy-and-pasting.

File: test/e2e/kubernetes/deployment/deployment.go [hunk 15]
```diff
@@ -212,7 +212,7 @@ func (d *Deployment) WaitForReplicas(n int, sleep, duration time.Duration) ([]po
 		select {
 		case err := <-errCh:
 			return pods, err
-		case _ = <-readyCh:
```
[the remark this answers is not in the document]
Comment (mboersma) [reply test/e2e/kubernetes/deployment/deployment.go]: I guess you can elide the no-op assignment inside a `select` statement.

File: pkg/api/convertertoagentpoolonlyapi_test.go (hunk 13 above)
Comment (mboersma) [reply pkg/api/convertertoagentpoolonlyapi_test.go]: It stops here.

File: pkg/acsengine/transform/apimodel_merger.go (hunk 3 above)
Comment (mboersma) [reply pkg/acsengine/transform/apimodel_merger.go]: `pkg/acsengine/transform/apimodel_merger.go:26:5:warning: should omit nil check; len() for nil slices is defined as zero (S1009) (gosimple)`.

**Pull request merged**